### PR TITLE
Created integration tests for WebConnectivity and API

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Part of https://github.com/ooni/backend/issues/450

### CODE REVIEW

**Comment by lorenzoPrimi** [comment]
Why the tests are not running?

File: Podfile [hunk 1]
```diff
@@ -15,5 +15,7 @@ target 'ooniprobe' do
     pod 'Countly'
 end
 target 'OONIProbeUnitTests' do
+    pod 'oonimkall', :podspec => 'https://dl.bintray.com/ooni/ios/oonimkall-2020.10.06-103613.podspec'
+    pod 'mkall', :git => 'https://github.com/measurement-kit/mkall-ios.git', :tag => 'v0.10.0'
```
Comment (bassosimone) [inline Podfile:19]:
```suggestion
```

File: ooniprobe.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1691,7 +1714,7 @@
 				STRIP_INSTALLED_PRODUCT = NO;
 				TARGETED_DEVICE_FAMILY = "1,2";
 				USER_HEADER_SEARCH_PATHS = "";
-				VALID_ARCHS = "arm64 armv7s i386 x86_64";
+				VALID_ARCHS = "armv7s armv7 i386 x86_64";
```
Comment (bassosimone) [inline ooniprobe.xcodeproj/project.pbxproj:1717]:
```suggestion
				VALID_ARCHS = "arm64 armv7s i386 x86_64";
```

File: ooniprobe.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1771,6 +1794,7 @@
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				TARGETED_DEVICE_FAMILY = "1,2";
 				TEST_HOST = "$(BUILT_PRODUCTS_DIR)/ooniprobe.app/ooniprobe";
+				VALID_ARCHS = "armv7s armv7 i386 x86_64";
```
Comment (bassosimone) [inline ooniprobe.xcodeproj/project.pbxproj:1797]:
```suggestion
				VALID_ARCHS = "arm64 armv7s i386 x86_64";
```

File: Podfile.lock [hunk 4]
```diff
@@ -6,6 +6,7 @@ PODS:
   - DZNEmptyDataSet (1.8.1)
   - lottie-ios (2.5.3)
   - MBProgressHUD (1.2.0)
+  - mkall (0.10.0)
```
Comment (bassosimone) [inline Podfile.lock:9]:
```suggestion
```

File: Podfile.lock [hunk 5]
```diff
@@ -23,6 +24,7 @@ DEPENDENCIES:
   - DZNEmptyDataSet
   - lottie-ios (= 2.5.3)
   - MBProgressHUD
+  - mkall (from `https://github.com/measurement-kit/mkall-ios.git`, tag `v0.10.0`)
```
Comment (bassosimone) [inline Podfile.lock:27]:
```suggestion
```

File: Podfile.lock [hunk 6]
```diff
@@ -45,13 +47,19 @@ SPEC REPOS:
     - XNGMarkdownParser
 
 EXTERNAL SOURCES:
+  mkall:
+    :git: https://github.com/measurement-kit/mkall-ios.git
+    :tag: v0.10.0
```
Comment (bassosimone) [inline Podfile.lock:52]:
```suggestion
```

File: Podfile.lock [hunk 7]
```diff
@@ -45,13 +47,19 @@ SPEC REPOS:
...
   oonimkall:
     :podspec: https://dl.bintray.com/ooni/ios/oonimkall-2020.10.06-103613.podspec
   SharkORM:
     :git: https://github.com/sharksync/sharkorm
     :tag: v2.3.67
 
 CHECKOUT OPTIONS:
+  mkall:
+    :git: https://github.com/measurement-kit/mkall-ios.git
+    :tag: v0.10.0
```
Comment (bassosimone) [inline Podfile.lock:62]:
```suggestion
```

File: Podfile.lock [hunk 8]
```diff
@@ -62,6 +70,7 @@ SPEC CHECKSUMS:
   DZNEmptyDataSet: 9525833b9e68ac21c30253e1d3d7076cc828eaa7
   lottie-ios: a50d5c0160425cd4b01b852bb9578963e6d92d31
   MBProgressHUD: 3ee5efcc380f6a79a7cc9b363dd669c5e1ae7406
+  mkall: 33f2dddb7e2d87a838d3b0e1403601c0f8d331d4
```
Comment (bassosimone) [inline Podfile.lock:73]:
```suggestion
```

File: Podfile.lock [hunk 9]
```diff
@@ -71,6 +80,6 @@ SPEC CHECKSUMS:
   TTTAttributedLabel: 8cffe8e127e4e82ff3af1e5386d4cd0ad000b656
   XNGMarkdownParser: aed98c14f0c0eb20064184ddf9bac26c482722b2
 
-PODFILE CHECKSUM: 9cb955898fdb8b26b1f47b3eada5645064cd2ca3
+PODFILE CHECKSUM: 17263fdd8bb5ec10e429117072c8ed578ce4699c
```
Comment (bassosimone) [inline Podfile.lock:83]:
```suggestion
PODFILE CHECKSUM: 9cb955898fdb8b26b1f47b3eada5645064cd2ca3
```

File: ooniprobe.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -1071,6 +1074,7 @@
 				EDF23EE72225258100B97520 /* Sources */,
 				EDF23EE82225258100B97520 /* Frameworks */,
 				EDF23EE92225258100B97520 /* Resources */,
+				3631AB52440CE392AE49C5FB /* [CP] Embed Pods Frameworks */,
```
Comment (bassosimone) [inline ooniprobe.xcodeproj/project.pbxproj:1077]:
```suggestion
```

File: ooniprobe.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -1252,6 +1256,24 @@
...
+			);
+			name = "[CP] Embed Pods Frameworks";
+			outputPaths = (
+				"${TARGET_BUILD_DIR}/${FRAMEWORKS_FOLDER_PATH}/mkall.framework",
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "\"${PODS_ROOT}/Target Support Files/Pods-OONIProbeUnitTests/Pods-OONIProbeUnitTests-frameworks.sh\"\n";
+			showEnvVarsInLog = 0;
+		};
```
Comment (bassosimone) [inline ooniprobe.xcodeproj/project.pbxproj:1276]:
```suggestion
```

File: ooniprobe/Utility/OONIApi.m [hunk 12]
```diff
@@ -4,10 +4,13 @@
 #import "NetworkSession.h"
 #import "Url.h"
 #import "VersionUtility.h"
-
+#define OONI_ORCHESTRATE_BASE_URL @"orchestrate.ooni.io"
+#define OONI_API_BASE_URL @"api.ooni.io"
```
Comment (bassosimone) [inline ooniprobe/Utility/OONIApi.m:8]:
```suggestion
#define OONI_API_BASE_URL @"api.ooni.io"

```

File: ooniprobe/Utility/OONIApi.m [hunk 13]
```diff
@@ -191,6 +213,15 @@ + (void)checkReportId:(NSString*)report_id
     [downloadTask resume];
 }
 
++ (void)checkReportId:(NSString*)report_id
+           onSuccess:(void (^)(BOOL))successcb
+             onError:(void (^)(NSError*))errorcb{
+    [self checkReportId:OONI_API_BASE_URL
```
Comment (bassosimone) [inline ooniprobe/Utility/OONIApi.m:219]: @lorenzoPrimi do we have a test for this function?

**Review by bassosimone — CHANGES_REQUESTED** [review]
Thanks for working on this! I've put together a set of suggestions that should help to simplify the final diff. Plus, I have also a clarifying question to make sure we are testing everything we need to test.

File: ooniprobe/Utility/OONIApi.m (hunk 13 above)
Comment (lorenzoPrimi) [reply ooniprobe/Utility/OONIApi.m:220]: No, I don't think is needed, was an hotfix to prevent to ddos ouselves again

Comment (bassosimone) [reply ooniprobe/Utility/OONIApi.m:220]: ok

**Review by bassosimone — APPROVED** [review]
🐳

**Pull request merged**